common/lib/capa/capa/tests/test_capa_problem.py

```diff
@@ -317,3 +326,123 @@ def test_label_attribute_mismatches_question_tag(self):
+            '//section[@class="wrapper-problem-response"]/div[@class="multi-inputs-group"]'
+        )
+        self.assertEqual(len(multi_inputs_group), 1)
+
+        # verify that multi input group label <p> tag exists and its
+        # id matches with correct multi input group aria-labelledby
+        multi_inputs_group_label_id = multi_inputs_group[0].attrib.get('aria-labelledby')
+        multi_inputs_group_label = html.xpath('//p[@id="{}"]'.format(multi_inputs_group_label_id))
+        self.assertEqual(len(multi_inputs_group_label), 1)
+        self.assertEqual(multi_inputs_group_label[0].text, group_label)
+
+        # verify that label for each input comes only once
+        for input_label in input_labels:
+            # normalize-space is used to remove whitespace around the text
+            input_label_element = multi_inputs_group[0].xpath('//*[normalize-space(text())="{}"]'.format(input_label))
+            self.assertEqual(len(input_label_element), 1)
+
+    def test_optionresponse(self):
+        """
+        Verify that optionresponse problem with multiple inputtypes is rendered correctly.
+        """
+        group_label = 'Choose the correct color'
+        input1_label = 'What color is the sky?'
+        input2_label = 'What color are pine needles?'
+        xml = """
+        <problem>
+            <optionresponse>
+                <label>{}</label>
+                <optioninput options="('yellow','blue','green')" correct="blue" label="{}"/>
+                <optioninput options="('yellow','blue','green')" correct="green" label="{}"/>
+            </optionresponse>
+        </problem>
+        """.format(group_label, input1_label, input2_label)
+        problem = self.capa_problem(xml)
+        self.assert_problem_html(problem.get_html(), group_label, input1_label, input2_label)
+
+    @ddt.unpack
+    @ddt.data(
+        {'inputtype': 'textline'},
+        {'inputtype': 'formulaequationinput'}
+    )
+    def test_customresponse(self, inputtype):
+        """
+        Verify that customresponse problem with multiple textline
+        and formulaequationinput inputtypes is rendered correctly.
+        """
+        group_label = 'Enter two integers that sum to 10.'
+        input1_label = 'Integer 1'
+        input2_label = 'Integer 2'
+        xml = textwrap.dedent("""
+        <problem>
+            <customresponse cfn="test_add_to_ten">
+        <script type="loncapa/python">
+        def test_add_to_ten(expect, ans):
+            return test_add(10, ans)
+        </script>
+                <label>{}</label>
+                <{inputtype} size="40" correct_answer="3" label="{}" /><br/>
+                <{inputtype} size="40" correct_answer="7" label="{}" />
+            </customresponse>
+        </problem>
+        """.format(group_label, input1_label, input2_label, inputtype=inputtype))
+        problem = self.capa_problem(xml)
+        self.assert_problem_html(problem.get_html(), group_label, input1_label, input2_label)
```

Comment:
@muhammad-ammar Can we add/update a test where we explicitly check that if there is single input in responsetype then there is no group label, group div and input labels.